tests/ga/test_extension.py

```diff
@@ -567,6 +573,299 @@ def test_ext_handler_reporting_status_file(self, *args):
+              "max_limit_percentage": 20,
+              "max_limit_MBs": 1000,
+              "memory_pressure_warning": "low",
+              "memory_oom_kill": "enabled"
+            }
+          }
+        }
+      }
+    }
+    '''
+        handler_config = HandlerConfiguration(json.loads(data))
+        self.assertIsNone(handler_config.get_cpu_limits_by_extension())
+
+        memory_limits = handler_config.get_memory_limits_by_extension()
+        self.assertEqual(memory_limits.max_limit_percentage, 20)
+        self.assertEqual(memory_limits.max_limit_MBs, 1000)
+        self.assertEqual(memory_limits.memory_pressure_warning, "low")
+        self.assertEqual(memory_limits.memory_oom_kill, "enabled")
+
+        data = '''{
+              "name": "ExampleHandlerLinux",
+              "version": 1.0,
+              "handlerConfiguration": {
+                "linux": {
+                  "resources": {
+                    "memory": {
+                      "max_limit_percentage": 20,
+                      "max_limit_MBs": 1000,
+                      "memory_pressure_warning": "low"
+                    }
+                  }
+                }
+              }
+            }
+            '''
+        handler_config = HandlerConfiguration(json.loads(data))
+        memory_limits = handler_config.get_memory_limits_by_extension()
+        self.assertEqual(memory_limits.max_limit_percentage, 20)
+        self.assertEqual(memory_limits.max_limit_MBs, 1000)
+        self.assertEqual(memory_limits.memory_pressure_warning, "low")
+        self.assertEqual(memory_limits.memory_oom_kill, "disabled")
+
+        data = '''{
+                      "name": "ExampleHandlerLinux",
+                      "version": 1.0,
+                      "handlerConfiguration": {
+                        "linux": {
+                          "resources": {
+                            "memory": {
+                              "max_limit_percentage": 20,
+                              "max_limit_MBs": 1000
+                            }
+                          }
+                        }
+                      }
+                    }
+                    '''
+        handler_config = HandlerConfiguration(json.loads(data))
+        memory_limits = handler_config.get_memory_limits_by_extension()
+        self.assertEqual(memory_limits.max_limit_percentage, 20)
+        self.assertEqual(memory_limits.max_limit_MBs, 1000)
+        self.assertIsNone(memory_limits.memory_pressure_warning)
+        self.assertEqual(memory_limits.memory_oom_kill, "disabled")
+
+    def test_handler_configuration_incorrect_memory_configuration(self, *args):
+        data = '''{
+      "name": "ExampleHandlerLinux",
+      "version": 1.0,
+      "handlerConfiguration": {
+        "linux": {
+          "resources": {
+            "memory": {
+              "max_limit_percentage": 20,
+              "max_limit_MBs": 1000,
+              "memory_pressure_warning": "low",
```

Comment:
can you give details (can be offline) of how the memory pressure warning and oom settings will work?